lib/core/src/Cardano/Wallet.hs

```diff
@@ -1500,28 +1508,23 @@ balanceTransaction
     generateChange
     (pp, nodePParams)
     ti
-    (internalUtxoAvailable, wallet, pendingTxs)
-    (PartialTx partialTx@(cardanoTx -> Cardano.InAnyCardanoEra _ (Cardano.Tx (Cardano.TxBody bod) _)) externalInputs redeemers)
+    (internalUtxoAvailable, wallet, _pendingTxs)
+    ptx@(PartialTx partialTx@(cardanoTx -> Cardano.InAnyCardanoEra _ (Cardano.Tx (Cardano.TxBody bod) _)) externalInputs redeemers)
     = do
-    let (outputs, txWithdrawal, txMetadata, txAssetsToMint, txAssetsToBurn)
-            = extractFromTx partialTx
-
     guardExistingCollateral
-    guardZeroAdaOutputs outputs
-    guardDeposits
+    guardZeroAdaOutputs (extractOutputsFromTx partialTx)
     guardConflictingWithdrawalNetworks
 
-    (delta, extraInputs, extraCollateral, extraOutputs) <- do
+    -- TODO: Nothing guarantees consistency between externalInputs and the
+    -- actual txins!
```

Comment:
Create a task for this

Reply:
Think through

Reply:
@Anviking I agree it's a good idea to make a task for this.

Do you think it qualifies as "technical debt"?

Regardless, perhaps after creating a task, we can make a "TODO" item in the code that references the task.

What do you think?

Reply:
:+1: 2ed7982740998eae1aca0a25701043389441daef and https://input-output.atlassian.net/browse/ADP-1544